game/src/ecs/components/HealthComponent.java

```diff
@@ -0,0 +1,178 @@
+     * @param getHitAnimation Animation to be played as the entity was hit
+     * @param dieAnimation Animation to be played as the entity dies
+     */
+    public HealthComponent(
+            Entity entity,
+            int maximalHitPoints,
+            IOnDeathFunction onDeath,
+            Animation getHitAnimation,
+            Animation dieAnimation) {
+        super(entity);
+        this.maximalHitPoints = maximalHitPoints;
+        this.currentHitPoints = maximalHitPoints;
+        this.onDeath = onDeath;
+        this.getHitAnimation = getHitAnimation;
+        this.dieAnimation = dieAnimation;
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Creates a HealthComponent with default values
+     *
+     * @param entity associated entity
+     */
+    public HealthComponent(@DSLContextMember(name = "entity") Entity entity) {
+        super(entity);
+        this.maximalHitPoints = 1;
+        this.currentHitPoints = 1;
+        this.onDeath = entity2 -> {};
+        this.getHitAnimation = new Animation(missingTexture, 100);
+        this.dieAnimation = new Animation(missingTexture, 100);
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Adds damage, which is accounted for by the system
+     *
+     * @param damage Damage that should be inflicted
+     */
+    public void getHit(Damage damage) {
+        damageToGet.add(damage);
+    }
+
+    /** Triggers the onDeath Function */
+    public void triggerOnDeath() {
+        onDeath.onDeath(entity);
+    }
+
+    /**
+     * @return List with all damage objects that still need to be accounted for
+     */
+    public List<Damage> getDamageList() {
+        return damageToGet;
+    }
+
+    /**
+     * Calculate the amount of damage of a certain type
+     *
+     * @param dt Type of damage object that still need to be accounted for
+     * @return Sum of all damage objects of type dt (default: 0)
+     */
+    public int getDamage(DamageType dt) {
+        return damageToGet.stream()
+                .filter(d -> d.damageType() == dt)
+                .mapToInt(Damage::damageAmount)
+                .sum();
+    }
+
+    /** Clear the damage list */
+    public void clearDamageList() {
+        damageToGet.clear();
+    }
+
+    /**
+     * Sets the current life points, capped at the value of the maximum hit-points
+     *
+     * @param amount new amount of current hit-points
+     */
+    public void setCurrentHitPoints(int amount) {
+        this.currentHitPoints = Math.min(maximalHitPoints, amount);
+    }
+
+    /**
+     * Sets the value of the Maximum Hit-points. If the new maximum hit-points are less than the
+     * current hit-points, the current points are set to the new maximum hit-points.
+     *
+     * @param amount new amount of maximal hit-points
+     */
+    public void setMaximalHitPoints(int amount) {
+        this.maximalHitPoints = amount;
+        currentHitPoints = Math.min(currentHitPoints, maximalHitPoints);
```

Comment:
Warum? Ist das nicht egal - wenn ich mehr Hit-Points habe, als ich darf, dann bin ich halt genauso tot als wenn ich "nur" Max-Hitpoints habe?

Und falls das doch wichtig ist: Dann sollte das im Methodennamen rauskommen, dass hier (a) die maximalen Hitpoints aktualisiert werden und (b) die vorhandenen Hitpoints normalisiert werden. Oder besser eine zweite Methode zum Normalisieren, die dann nach setMaxHitP aufgerufen werden soll?

Reply:
Hitpoints sind Lebenspunkte.

Reply:
Ah, ja. Passt dann der Name?

Reply:
Ja. 
Kommt grob aus "Anzahl der Treffer die du überlebst". 
Ist gängige Spielsprache. 


https://de.wikipedia.org/wiki/Trefferpunkt

Reply:
Ja, my bad. Dann gleich die nächste Frage: Wenn "HitPoints" die Anzahl der Treffer sind, die ich verkrafte (oder eher die Menge an Damage, oder?) - warum muss ich ein "maxHitPoints" und "currentHitPoints" halten? Das wäre doch immer das selbe? Irgendwas verstehe ich noch nicht.

Reply:
Naja, typischer Wiese hast du ja 10/10 Lebenspunkte. Dann kriegst du auf die Schnauze und hast 8/10 Lebenspunkten. 
Wenn du jetzt einen Heiltrank trinkst, sollst du nicht mehr als deine Ursprünglichen /10 Lebenspunkte bekommen. 
Deine aktuellen Lebenspunkte sind daher, wie der name sagt, was du JETZT grade noch hast. Die maximalen Lebenspunkte sind, wie die name schon sagt, die Lebenspunkte die du MAXIMAL haben kannst. Also egal wie viele Heiltränke du trinkst, du wirst nie über 10/10 hinauskommen.
Außer du findest einen super duper Trank, der deine maximalen Lebenspunkte erhöht. 

Ist auch. gängig in jeder Art von Spiel.

Reply:
Ja - aber wo ist dann der Unterschied zu den HP (HealthPoints)?

Reply:
Kurz gesagt, es gibt keinen Unterschied zwischen HitPoints und HealthPoints. Außer dass HitPoints der erste Name für die Lebenspunkte waren und später durch Lebensbalken und andere Anzeigen zu HealthPoints genannt wurden, damit der Name an sich gleich klingt. Und somit die Hitbar welche bei Reaktionsevents existiert, nicht gleich klingt wie die Hitbar für die HitPoints.

Und bei uns im Deutschen klingt Lebenspunkte besser, da sie die Lebensenergie repräsentieren und Trefferpunkte eher selten verwendet werden. Im Englischen klingt für mich die Kombination HitPoints und LifeEnergy besser als HealthPoints und LifeEnergy.  

Weiß nicht, ob das vielleicht dir hilft. Also ist es einfach reine Formalität, wofür man sich entscheidet.

Und wenn wir jetzt das letzte mit in den Raum werfen, Schadenspunkte und DamagePoints welche sagen wie viel von der Lebensenergie abgezogen wird.

Reply:
Prima, dann ist das geklärt :)

@AMatutat Du merkst, dass ich drüber gestolpert bin und sich hier eine eigentlich nicht notwendige Diskussion entsponnen hat. Prinzipiell ist mir das ziemlich egal, ob das "health*" oder "hit*" oder wie heisst - aber könnt ihr das bitte _vereinheitlichen_? Also wenn die Systeme und die Component "health" heissen, dann wäre das doch auch der Präfix für die tatsächlichen und maximalen Punkte und die entsprechenden Methoden?

Reply:
Ja. Habs angepasst.